libafl/src/observers/cmp.rs

```diff
@@ -255,27 +270,244 @@ where
 impl<'a, CM, S> StdCmpObserver<'a, CM, S>
 where
     CM: CmpMap + Serialize + DeserializeOwned,
-    S: UsesInput,
+    S: UsesInput + HasMetadata,
 {
     /// Creates a new [`StdCmpObserver`] with the given name and map.
     #[must_use]
-    pub fn new(name: &'static str, map: &'a mut CM) -> Self {
+    pub fn new(name: &'static str, map: &'a mut CM, add_meta: bool) -> Self {
```

Comment:
Maybe a new method instead of a `bool`? Or at least explain in the docstring what it does

Reply:
Another method is fine, with add_meta true by default for new()